assets/src/apps/authoring/components/PropertyEditor/custom/ColorPickerWidget.tsx

```diff
@@ -1,24 +1,41 @@
-import React, { CSSProperties } from 'react';
-import { useState } from 'react';
+import React, { CSSProperties, useRef, useState } from 'react';
 import { SketchPicker } from 'react-color';
 interface ColorPickerProps {
   value: any;
 }
 
 const getColorValueString = (value: any) => {
-  return `rgba(${value.rgb.r},${value.rgb.g},${value.rgb.b},${value.rgb.a})`;
+  return (
+    value.rgb.a && value.rgb.a != 0 ?
+      `rgba(${value.rgb.r},${value.rgb.g},${value.rgb.b},${value.rgb.a})`
+      :
+      `rgb(${value.rgb.r},${value.rgb.g},${value.rgb.b})`
+  );
 }
 
 const getRGBColorValue = (value: any) => {
   if (value) {
-    const parts = value.replace('rgba(', '').replace(')', '').split(',');
-    return { r: parts[0], g: parts[1], b: parts[2], a: parts.length > 3 ? parts[3] : 1 }
+    const parts = value.replace('rgba(', '').replace('rgb(','').replace(')', '').split(',');
+    return { r: parts[0], g: parts[1], b: parts[2], a: parts.length > 3 ? parts[3] : 100 }
   }
-  return { r: 255, g: 255, b: 255, a: 0.0 };
+  return { r: 255, g: 255, b: 255, a: 0 };
 }
+
 const ColorPickerWidget: React.FC<ColorPickerProps> = (props: any) => {
   const color = getRGBColorValue(props.value);
   const [displayPicker, setDisplayPicker] = useState(false);
+
+  const handleColorBoxClick = () => {
+    setDisplayPicker(true);
+    document.addEventListener("mousedown", handleClick);
+  };
+  const handleClick = (event: any) => {
+    if (pickerRef.current && !pickerRef.current.contains(event.target)) {
+      setDisplayPicker(false);
+      document.removeEventListener("mousedown", handleClick);
+    }
+  }
```

Comment:
missing semi; also the whole file looks like it could probably use a formatter pass; (it should add all missing semi for you)